docs/DeveloperGuide.md

```diff
@@ -5,9 +5,147 @@
+
+- Step 3: The created `HydrationEntry` instance is added into the `ArrayList<Entry>` attribute of the `HydrationList`.
+
+### Calculating hydration requirements for each user
+
+#### Design Considerations
+
+**General Health Guidelines:** The recommended daily intake of water for an average adult is around 8 glasses or approximately 2000 milliliters. This guideline is commonly recommended by health authorities and organizations such as HealthHub.
+
+**Ease of Implementation:** Setting a standard hydration requirement simplifies the tracking process for users. It provides a clear goal to strive for, making it easier for individuals to monitor and maintain their hydration levels consistently.
+### Hydration list feature
+
+The `hydration list` feature lists out the record of all the Hydration data that the user has keyed in. The Hydration data are all stored into a `ArrayList<Entry> hydrationArrayList` attribute of the `HydrationList` Class. Hydration data are printed when the `printHydrationList()` function is called.
+
+The `printHydrationList()` function iterates through the `hydrationArrayList` and prints out the Entries according to its order in the Array List.
+
+The Class diagram for Hydration list feature is shown below. Unrelated attributes and Classes were excluded.
+
+![HydrationListClassDiagram.png](HydrationListClassDiagram.png)
+
+### Hydration delete feature
+
+The `hydration delete` feature can delete the hydration record at specific index of hydration list. This functionality is facilitated by `HydrationList`. It implements one operation, namely:
+- `deleteEntry(String line)`
+
+Given below is an example usage scenario and how this mechanism behaves at every step:
+- Step 1: When the user inputs the command `hydration delete INDEX` in the terminal, the string is sent to `Ui#handleUserInput()`, which will call `Ui#handleHydrationInput()`.
+
+- Step 2: After the `Ui#handleHydrationInput()` matching `delete hydration` key word, the string will be passed into deleteEntry(String line) to execute delete process.
+
+- Step 3: The string will be divided to two substrings according to the command syntax. Index will be tried to get from the second substring by `Integer.parseInt()`.
+
+- Step 4: The hydration record (`Entry`) stored in the `ArrayList<Entry> hydrationList` will be deleted by calling `hydrationArrayList.remove((index-1));` and a successful deleting message will be shown in terminal by calling `HydrationListUi#successfulDeletedMessage(toDelete)`
+
+- Step 5: The latest hydration list will be updated to saving file by calling `HydrationList#updateFile()`.
+
+The Sequence diagram for Hydration delete feature is shown below:
+![HydrationDeleteDiagram.png](HydrationDeleteDiagram.png)
```

Comment:
The notation for this sequence diagram appears to be different from the rest of the diagrams. The methods called seem to be indicated with the class. E.g. Ui#handleHydrationinput. Would it be better to standardize the notations from the previous diagrams, in particular, following the notation taught during lectures?